managed/CounterStrikeSharp.API/Core/BasePlugin.cs

```diff
@@ -162,6 +163,16 @@ public void AddCommand(string name, string description, CommandInfo.CommandCallb
 
                 var entity = new CCSPlayerController(NativeAPI.GetEntityFromIndex(i + 1));
                 var command = new CommandInfo(ptr, entity);
+
+                // Do not execute command if we do not have the correct permissions.
+                var methodInfo = handler?.GetMethodInfo();
+                var attr = methodInfo?.GetCustomAttribute(typeof(PermissionHelperAttribute), true) as PermissionHelperAttribute;
```

Comment:
You can use the typesafe version of custom attribute using `.GetCustomAttribute<PermissionHelperAttribute>()?.RequiredPermissions`